for v3.4.0  canopennode with sensor conflicting types for 'float64_t'

## base on v3.4.0 
```diff --git a/samples/modules/canopennode/prj.conf b/samples/modules/canopennode/prj.conf
index 905a1934f4..45bb026868 100644
--- a/samples/modules/canopennode/prj.conf
+++ b/samples/modules/canopennode/prj.conf
@@ -1,5 +1,6 @@
 CONFIG_LOG=y
 CONFIG_CANOPEN_LOG_LEVEL_DBG=y
+CONFIG_SENSOR=y

 CONFIG_GPIO=y

diff --git a/samples/modules/canopennode/src/main.c b/samples/modules/canopennode/src/main.c
index 595683e13b..9f316d153c 100644
--- a/samples/modules/canopennode/src/main.c
+++ b/samples/modules/canopennode/src/main.c
@@ -9,6 +9,7 @@
 #include <zephyr/sys/reboot.h>
 #include <zephyr/settings/settings.h>
 #include <canopennode.h>
+#include <zephyr/drivers/sensor.h>

 #define LOG_LEVEL CONFIG_CANOPEN_LOG_LEVEL
 #include <zephyr/logging/log.h>
```

```
cd samples/modules/canopennode/
west build -b stm32f3_disco ./
```

```
dir/src/main.c.obj -c /opt/nordic/ncs/v2.3.0/zephyr/samples/modules/canopennode/src/main.c
In file included from /opt/nordic/ncs/v2.3.0/zephyr/include/zephyr/drivers/sensor.h:26,
                 from /opt/nordic/ncs/v2.3.0/zephyr/samples/modules/canopennode/src/main.c:12:
/opt/nordic/ncs/v2.3.0/zephyr/include/zephyr/dsp/types.h:61:16: error: conflicting types for 'float64_t'; have 'double'
   61 | typedef double float64_t;
      |                ^~~~~~~~~
In file included from /opt/nordic/ncs/v2.3.0/modules/lib/canopennode/CO_driver.h:35,
                 from /opt/nordic/ncs/v2.3.0/modules/lib/canopennode/CANopen.h:67,
                 from /opt/nordic/ncs/v2.3.0/zephyr/modules/canopennode/./canopennode.h:23,
                 from /opt/nordic/ncs/v2.3.0/zephyr/samples/modules/canopennode/src/main.c:11:
/opt/nordic/ncs/v2.3.0/zephyr/modules/canopennode/./CO_driver_target.h:50:23: note: previous declaration of 'float64_t' with type 'float64_t' {aka 'long double'}
   50 | typedef long double   float64_t;
      |                       ^~~~~~~~~
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /opt/nordic/ncs/v2.3.0/zephyr/samples/modules/canopennode/build
```

--- a/modules/canopennode/CO_driver_target.h
+++ b/modules/canopennode/CO_driver_target.h
@@ -22,6 +22,7 @@ extern "C" {
 #include <zephyr/types.h>
 #include <zephyr/device.h>
 #include <zephyr/toolchain.h>
+#include <zephyr/dsp/types.h> /* float32_t, float64_t */
 
 /* Use static variables instead of calloc() */
 #define CO_USE_GLOBALS
@@ -46,11 +47,12 @@ extern "C" {
 #endif
 
 typedef bool          bool_t;
-typedef float         float32_t;
-typedef long double   float64_t;
 typedef char          char_t;
 typedef unsigned char oChar_t;
 typedef unsigned char domain_t;
+
+BUILD_ASSERT(sizeof(float32_t) >= 4);
+BUILD_ASSERT(sizeof(float64_t) >= 8);
 
 typedef struct canopen_rx_msg {
 	uint8_t data[8];

